Petition filename updates

robert-w-gries opened the pull request:
Resolves #546, Resolves #548, Resolves #549

Request:

When the petitions currently download, they are named with date formatted in MM-DD-YYYY. Our files are supposed to be named with date formatted as YYYY-MM-DD. Any chance we can adjust the tool so the petitions download with date formatted as YYYY-MM-DD instead? If not, it’s not a big deal, we can always rename the files. But if it’s an easy change, it would be great!

1. Add the word DRAFT between the statute number and client name?
2. For form AOC-CR-297, can an F be added to 145.5?
3. For form AOC-CR-298, can an M be added to 145.5? That would delineate which one is expunging felonies and which one is expunging misdemeanors.

  For example, a petition to expunge felony convictions in Wake Superior Court would download as follows:
  
  2025-06-12 Wake SC 145.5F DRAFT John Doe

 
4. For form AOC-CR-287 (dismissals), can the statute number be just 146 instead of 146(a) or 146(a1)?

5. For form AOC-CR-288 (not guiltys), can the statute number be 146NG instead of 146(a2)?

dulshen reviewed (commented):
The changes related to Emily's requested changes all looked good to me.

For the change to the response type for deleting a batch: I'm not exactly sure what the issue was here, but was there no good way to update front end to accept the 204 response?

Also if not, would this issue in the frontend also apply to other viewsets in the backend that use the default destroy action (which appears to also give 204 response)? (e.g. deleteAgency)

robert-w-gries commented:
Deleting batches would hang in the frontend because we had a body provided in the 204 no content response (Django would fail to send the response I think). I personally didn't care which response type it was, but I can see the value in keeping it consistent across endpoints. I've updated the response to use 204 status and removed the `{}` to fix the issue with the delay.
